### PR TITLE
Always send the Date HTTP header on API requests

### PR DESCRIPTION
**Pull request by Dantemss**

### CODE REVIEW

File: app/controllers/openstax/api/v1/api_controller.rb [hunk 1]
```diff
@@ -46,8 +48,12 @@ def session_user?
           !!current_session_user && doorkeeper_token.blank?
         end
 
+        def set_date_header
+          response.date = Time.now unless response.date?
```
Comment (jpslav) [inline app/controllers/openstax/api/v1/api_controller.rb]: this works?  is this the cache control header?  ok to use it for this purpose (other than making a different custom header field?)

**Comment by jpslav** [comment]
Can you add a spec?

File: app/controllers/openstax/api/v1/api_controller.rb (hunk 1 above)
Comment (Dantemss) [inline app/controllers/openstax/api/v1/api_controller.rb]: This is the purpose of this header according to http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18
Also according to the RFC, it should pretty much always be set (idk why Rails/thin don't do it).

**Comment by Dantemss** [comment]
Sure.

File: app/controllers/openstax/api/v1/api_controller.rb (hunk 1 above)
Comment (Dantemss) [inline app/controllers/openstax/api/v1/api_controller.rb]: Rails actually does set it here: https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/conditional_get.rb#L209

But only if you call `expires_in` in the controller.

**Comment by Dantemss** [comment]
Specs added.

**Pull request merged**